runtime/shared_common/j9shr.tdf

```diff
@@ -2919,3 +2919,25 @@ TraceEvent=Trc_SHR_INIT_j9shr_init_Entry Overhead=1 Level=6 Template="INIT enter
+TraceException=Trc_SHR_INIT_fetchStartupHintsFromSharedCache_Null_key Overhead=1 Level=1 Template="INIT::fetchStartupHintsFromSharedCache() : Failed to generate the key"
+TraceException=Trc_SHR_INIT_fetchStartupHintsFromSharedCache_Mem_Alloc_Failed Overhead=1 Level=1 Template="INIT::fetchStartupHintsFromSharedCache() : Failed to allocate memory"
+TraceEntry=Trc_SHR_INIT_storeStartupHintsToSharedCache_Entry Overhead=1 Level=2 Template="INIT::storeStartupHintsToSharedCache() : Entered"
+TraceEvent=Trc_SHR_INIT_storeStartupHintsToSharedCache_Set_Flags Overhead=1 Level=2 Template="INIT::storeStartupHintsToSharedCache() : dataDescriptor.flags is set to %zu"
+TraceException=Trc_SHR_INIT_storeStartupHintsToSharedCache_Store_Failed Overhead=1 Level=2 Template="INIT::storeStartupHintsToSharedCache() : Failed to store the hints to the shared cache"
+TraceExit=Trc_SHR_INIT_storeStartupHintsToSharedCache_Store_Successful Overhead=1 Level=2 Template="INIT::storeStartupHintsToSharedCache() : Successfully stored the hints to the shared cache"
+TraceExit=Trc_SHR_INIT_storeStartupHintsToSharedCache_Store_Nothing Overhead=1 Level=2 Template="INIT::storeStartupHintsToSharedCache() : Exit with nothing to store"
+TraceEntry=Trc_SHR_INIT_j9shr_storeGCHints_Entry Overhead=1 Level=2 Template="INIT::j9shr_storeGCHints() : Enter (heapSize1=%zu heapSize2=%zu forceReplace=%d)"
+TraceEvent=Trc_SHR_INIT_j9shr_storeGCHints_Overwirte_LocalHints Overhead=1 Level=2 Template="INIT::j9shr_storeGCHints() : Overwrite localStartupHints with heapSize1=%zu heapSize2=%zu"
+TraceEvent=Trc_SHR_INIT_j9shr_storeGCHints_Write_To_LocalHints Overhead=1 Level=2 Template="INIT::j9shr_storeGCHints() : Write heapSize1=%zu heapSize2=%zu to localStartupHints"
+TraceExit-Exception=Trc_SHR_INIT_j9shr_storeGCHints_Exit Overhead=1 Level=1 Template="INIT::j9shr_storeGCHints() : localStartupHints is NULL"
+TraceEntry=Trc_SHR_INIT_j9shr_findGCHints Overhead=1 Level=1 Template="INIT::j9shr_findGCHint() : Enter"
+TraceEvent=Trc_SHR_INIT_j9shr_findGCHints_Event_Found Overhead=1 Level=1 Template="INIT::j9shr_findGCHints() : GC hints found (heapSize1: %zu, heapSize2: %zu)"
+TraceExit=Trc_SHR_INIT_j9shr_findGCHints_Exit Overhead=1 Level=1 Template="INIT::j9shr_findGCHints() : Exit - returnVal %zd"
+TraceEvent=Trc_SHR_CM_storeSharedData_OverwriteExisting Overhead=1 Level=2 Template="CM storeSharedData: Existing data in the shared cache has been overwritten (result %p, data->address %p, foundDatalen %zu)."
+TraceEvent=Trc_SHR_CM_updateLocalHintsData_OverwriteHeapSizes Overhead=1 Level=2 Template="CM updateLocalHintsData: Existing hints (heapSize1=%zu, heapSize2=%zu) in the shared cache will be overwritten to (heapSize1=%zu, heapSize2=%zu)."
+TraceEvent=Trc_SHR_CM_updateLocalHintsData_WriteHeapSizes Overhead=1 Level=2 Template="CM updateLocalHintsData: Will write hints (heapSize1=%zu, heapSize2=%zu) to shared cache."
```

Comment:
I think we should redo the tracepoints to have fewer enabled by default. I'd like the bare minimum enabled that shows the fetched hint and the stored hint. Others can be a higher level not enabled by default. Because we want the Exceptions to be level 1 (or 2), it may be better not to have a TraceEntry (because entries and exits must be matched). If you want to keep the TraceEntry, better to have a single TraceExit that occurs regardless, and separate Exception tracepoints.

Reply:
I've removed the entry/exit trace points. The exceptions are kept level 1 or 2. Trc_SHR_INIT_fetchStartupHintsFromSharedCache_Hints_Found is level 1 and Trc_SHR_INIT_storeStartupHintsToSharedCache_Store_Successful is level 2. All other trace points are set to level 4.